### PR TITLE
First implementation of the broken line fit

### PR DESCRIPTION
**Pull request by fwyzard**
Merge the work by @GimmyTomas into CMSSW_10_6_X_Patatrack:

> Create modifiers for enabling the broken line fit on the cpu and on the gpu.
>
> Use dinamically-sized-matrices: the advantage over statically-sized ones is that the code would also work with n>4); the switch can be easily done at the start of the file.
>
> Update Eigen tests with the features used by the broken line fit.

### CODE REVIEW

**Comment by fwyzard** [comment]
Split #324, part 3 of N.

This PR *does not* include the work on `RecoPixelVertexing/PixelTrackFitting/test/testEigenGPU.cu` , due to the extensive conflicts. An updated version will be recovered the integration of #324 proceeds (unless @rovere feels like looking into it for historical records).

File: RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 #include "FitResult.h"
 
+#ifndef RFIT_DEBUG
+#define RFIT_DEBUG 0
+#endif  // RFIT_DEBUG
```
Comment (makortel) [inline RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h]: Is this `#define` something to be kept? (just asking)

Comment (fwyzard) [reply RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h]: it seems to be still used also in the final version, #338 .

**Comment by fwyzard** [comment]
`RecoPixelVertexing/PixelTrackFitting/test/testEigenGPU.cu` is thoroughly broken also in all but the last commit of #338, so it might be worth actually fixing it here.

File: RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h (hunk 1 above)
Comment (rovere) [reply RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h]: Ciao Matti,
I still think this `#define` is useful. There are a few aspects that should be better understood in both fits and having the ability to quickly dump information is valuable, to me.

Comment (makortel) [reply RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h]: That's fine. I just wanted confirm that having some debug stuff turned on by default is intentional.

Comment (fwyzard) [reply RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h]: Hi Matti,
why do you say

> turned on by default

?

Comment (makortel) [reply RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h]: These lines defines `RFIT_DEBUG` macro if it was not defined before, i.e. the macro is always defined. Later in the file only the existence of the macro is checked with `#ifdef RFIT_DEBUG`.

Comment (fwyzard) [reply RecoPixelVertexing/PixelTrackFitting/interface/RiemannFit.h]: Ah... I assumed we were using `#if RFIT_DEBUG`, not `#ifdef RFIT_DEBUG`.

**Pull request merged**